AutomatedLabTest/functions/Invoke-LabPester.ps1

```diff
@@ -21,37 +20,39 @@
         $OutputFile
     )
 
-    process
-    {
-        if (-not $Lab)
-        {
+    process {
+        if (-not $Lab) {
             $Lab = Import-Lab -Name $LabName -ErrorAction Stop -NoDisplay -NoValidation -PassThru
         }
 
         $global:pesterLab = $Lab # No parameters in Pester v5 yet
         $configuration = [PesterConfiguration]::Default
-        $configuration.Run.Path = Join-Path -Path $PSCmdlet.MyInvocation.MyCommand.Module.ModuleBase -ChildPath 'tests'
+        $config_paths = @() # Setup the array to pass to $Configuration.Run.Path
+        $config_paths += Join-Path -Path $PSCmdlet.MyInvocation.MyCommand.Module.ModuleBase -ChildPath 'tests' # This is for the built-in Roles
         $configuration.Run.PassThru = $PassThru.IsPresent
         [string[]]$tags = 'General'
         
-        if ($Lab.Machines.Roles.Name)
-        {
+        if ($Lab.Machines.Roles.Name) {
             $tags += $Lab.Machines.Roles.Name
         }
-        if ($Lab.Machines.PostInstallationActivity | Where-Object IsCustomRole)
-        {
+        if ($Lab.Machines.PostInstallationActivity | Where-Object IsCustomRole) {
             $tags += ($Lab.Machines.PostInstallationActivity | Where-Object IsCustomRole).RoleName
+            foreach ( $role in (($Lab.Machines.PostInstallationActivity | Where-Object IsCustomRole).RoleName)) {
+                $config_paths += Join-Path -Path $global:LabSources\CustomRoles -ChildPath $role
+            }
         }
-        if ($Lab.Machines.PreInstallationActivity | Where-Object IsCustomRole)
-        {
+        if ($Lab.Machines.PreInstallationActivity | Where-Object IsCustomRole) {
             $tags += ($Lab.Machines.PreInstallationActivity | Where-Object IsCustomRole).RoleName
+            foreach ( $role in (($Lab.Machines.PreInstallationActivity | Where-Object IsCustomRole).RoleName)) {
+                $config_paths += Join-Path -Path $global:LabSources\CustomRoles -ChildPath $role
+            }
         }
-
+        $configuration.Run.Path = $config_paths
```

Comment:
```suggestion
        $configuration.Run.Path = $config_paths | Sort-Object -Unique
```

Just a small suggestion - it should not occur, but it is still possible to add the same pre- and post-install activity. Since we always test at the end, we don't really need to run these twice :)